test/test-parse-statement.cpp

```diff
@@ -593,6 +606,44 @@ TEST_F(Test_Parse_Statement, if_with_else) {
                               "visit_variable_use",               // b
                           }));
   }
+
+  {
+    // if (x)
+    //   if (y)
+    //     z();
+    // else
+    //   w();
+    test_parse_and_visit_statement(
+        u8"if(x)\n\tif (y)\n\t\tz();\nelse\n\tw();"_sv,  //
+        u8"         ^^ Diag_Misleading_Braceless_If_Else_Indentation.if_span\n"_diag  //
+        u8"                           ^^^^ .else_span"_diag);
+
+    // if (x)
+    //   if (y)
+    //     z();
+    //   else
+    //     w();
+    test_parse_and_visit_statement(
+        u8"if(x)\n\tif (y)\n\t\tz();\n\telse\n\t\tw();"_sv, no_diags);
```

Comment:
For the no_diag tests, we can split the string onto multiple source lines:
```suggestion
    test_parse_and_visit_statement(
        u8"if(x)\n"_sv
        u8"\tif (y)\n"_sv
        u8"\t\tz();\n"_sv
        u8"\telse\n"_sv
        u8"\t\tw();"_sv,
        no_diags);
```